lib/cart/cart_repository.dart

```diff
@@ -1,29 +1,68 @@
 import 'package:flex_storefront/cart/apis/cart_api.dart';
 import 'package:flex_storefront/cart/apis/cart_exceptions.dart';
-import 'package:flex_storefront/cart/models/cart.dart';
 import 'package:flex_storefront/cart/models/cart_message.dart';
+import 'package:flex_storefront/cart/models/cart.dart';
 import 'package:flex_storefront/product_list/models/product.dart';
+import 'package:loggy/loggy.dart';
 import 'package:rxdart/subjects.dart';
+import 'package:shared_preferences/shared_preferences.dart';
 
-const kTestCart = '0cb93e1c-7fba-4b2e-9554-045d37979d78';
+mixin CartRepositoryLoggy implements LoggyType {
+  @override
+  Loggy<CartRepositoryLoggy> get loggy =>
+      Loggy<CartRepositoryLoggy>('CartRepositoryLoggy');
+}
 
-class CartRepository {
+class CartRepository with CartRepositoryLoggy {
   CartRepository({
     required CartApi cartApi,
   }) : _cartApi = cartApi {
     init();
   }
 
   final CartApi _cartApi;
+  late final SharedPreferences _prefs;
 
   final _cartStreamController = BehaviorSubject<Cart>();
   final _cartMessageStreamController =
       BehaviorSubject<CartMessage>.seeded(CartInitialize());
 
+  // Provide a [Stream] of the near real-time cart
+  Stream<Cart> getCartStream() => _cartStreamController.asBroadcastStream();
+
+  // Provide a [Stream] of important global cart messages
+  Stream<CartMessage> getCartMessageStream() =>
+      _cartMessageStreamController.asBroadcastStream();
+
+  String cartId = '';
+  Cart get currentCart => _cartStreamController.value;
+  bool get hasCart => _cartStreamController.hasValue;
+
+  /// Initialize the CartRepository, fetch the last-used cart from
+  /// local storage or create a new one to prepare the Cart for usage.
   void init() async {
-    // TODO: fetch previous carts from local storage
+    // 1. check local storage
+    _prefs = await SharedPreferences.getInstance();
+    cartId = _prefs.getString('anonymous_cart_guid') ?? '';
+
+    loggy.info(
+      'Cart initialization started, local storage cartId: $cartId',
+    );
+
+    if (cartId.isNotEmpty) {
+      await fetchCart();
+    } else {
+      await createCart();
+    }
+
+    loggy.info(
+      'Cart initialization ended, cart ready',
+    );
+  }
+
+  Future<void> fetchCart() async {
     try {
-      final cart = await _cartApi.fetchCart(cartCode: kTestCart);
+      final cart = await _cartApi.fetchCart(cartCode: cartId);
```

Comment:
This `cartId` should be read from storage here.

This another `cartId` is saved _after_ `init()` is called, this method won't see it.